components-core/src/main/java/com/adyen/checkout/components/core/internal/util/ValidationUtils.kt

```diff
@@ -55,25 +45,20 @@ object ValidationUtils {
      * Check if the Client Key is valid.
      *
      * @param clientKey A string to check if is a client key.
-     * @return If we consider it a valid client key or not.
-     */
-    fun isClientKeyValid(clientKey: String): Boolean {
-        return CLIENT_KEY_PATTERN.matcher(clientKey).matches()
-    }
-
-    /**
-     * Check if the Client Key matches the Environment
-     *
-     * @param clientKey A string to check if is a client key.
      * @param environment Selected environment in the configuration.
-     * @return If Client Key is preceded with correct prefix
+     * @return If the client key is valid and matches the chosen environment.
      */
-    fun doesClientKeyMatchEnvironment(clientKey: String, environment: Environment): Boolean {
-        val isTestEnvironment = environment == Environment.TEST
-        val isLiveEnvironment = LIVE_ENVIRONMENTS.contains(environment)
+    internal fun isClientKeyValid(clientKey: String, environment: Environment): Boolean {
+        return when (environment) {
+            Environment.TEST -> TEST_CLIENT_KEY_PATTERN.matcher(clientKey).matches()
+
+            Environment.APSE,
+            Environment.AUSTRALIA,
+            Environment.EUROPE,
+            Environment.INDIA,
+            Environment.UNITED_STATES -> LIVE_CLIENT_KEY_PATTERN.matcher(clientKey).matches()
 
-        return (isLiveEnvironment && clientKey.startsWith(CLIENT_KEY_LIVE_PREFIX)) ||
-            (isTestEnvironment && clientKey.startsWith(CLIENT_KEY_TEST_PREFIX)) ||
-            (!isLiveEnvironment && !isTestEnvironment)
+            else -> GENERIC_CLIENT_KEY_PATTERN.matcher(clientKey).matches()
```

Comment:
Should we even check this case? Can't think of a situation where a client key wouldn't be live or test 🤔

Reply:
Yeah I was in between with this one. Internally we have other environments as well, and although they are not accessible publicly we might want to use them ourselves, so instead of always returning valid or invalid I thought of adding this extra validation. WDYT?

Reply:
Hmm I think it's better to not build in a potential backdoor. When we need this for testing we can disable validation locally

Reply:
Agreed, fixed now